mono_repo/lib/src/commands/travis.dart

```diff
@@ -272,10 +273,34 @@ List<String> _calculateOrderedStages(Iterable<PackageConfig> configs) {
     }
   }
 
-  return components.map((c) => c.first).toList().reversed.toList();
+  var conditionalStages = Map<String, ConditionalStage>.from(
+      rootConfig.monoConfig.conditionalStages);
+
+  var orderedStages = components
+      .map((c) {
+        var stageName = c.first;
+
+        var matchingStage = conditionalStages.remove(stageName);
+        if (matchingStage != null) {
+          return matchingStage.toJson();
+        }
+
+        return stageName;
+      })
+      .toList()
+      .reversed
+      .toList();
+
+  if (conditionalStages.isNotEmpty) {
+    throw UserException('Error parsing mono_repo.yaml',
+        details: 'Stage `${conditionalStages.keys.first}` does not exist in '
+            'any mono_pkg.yaml files.');
```

Comment:
nit: `... but it was referenced in the root mono_repo.yaml file`

Reply:
ack